serialization/homework.py

```diff
@@ -2,18 +2,398 @@
 Для попереднього домашнього завдання.
 Для класу Колекціонер Машина і Гараж написати методи, які створюють інстанс обєкту
 з (yaml, json, pickle) файлу відповідно
-
 Для класів Колекціонер Машина і Гараж написати методи, які зберігають стан обєкту в файли формату
 yaml, json, pickle відповідно.
-
 Для класів Колекціонер Машина і Гараж написати методи, які конвертують обєкт в строку формату
 yaml, json, pickle відповідно.
-
 Для класу Колекціонер Машина і Гараж написати методи, які створюють інстанс обєкту
 з (yaml, json, pickle) строки відповідно
-
-
 Advanced
 Добавити опрацьовку формату ini
+"""
+
+from __future__ import annotations
+import codecs
+import pickle
+import random
+import uuid
+from builtins import str
+from collections import OrderedDict
+
+from constants import CARS_TYPES, CARS_PRODUCER, TOWNS
+import json
+from ruamel.yaml import YAML
+
+yaml = YAML()
+
+
+class Cesar:
+    def __init__(self, name: str, garages=[], register_id=uuid.uuid4()):
+        self.name = name
+        self.garages = garages
+        self.register_id = uuid.uuid4()
+        self.yaml = YAML()
+
+    def __repr__(self):
+        return 'name: {self.name}, garage: {self.garages}, register_id: {self.register_id}'.format(self=self)
+
+    def to_json(obj: Cesar):
+        garages = [garage.to_json_string() for garage in obj.garages]
+        data = {"name": obj.name,
+                "garages": garages,
+                "register_id": str(obj.register_id)}
+        return data
+
+    @staticmethod
+    def from_json(data):
+        name = data["name"]
+        register_id = data["register_id"]
+        garages1 = [Garage.from_json_string(garage) for garage in data['garages']]
+        return Cesar(name=name, garages=garages1, register_id=register_id)
+
+    def to_json_string(self):
+        cesar_json_string = json.dumps(self, default=Cesar.to_json, indent=6)
+        return str(cesar_json_string)
+
+    @staticmethod
+    def from_json_string(cesar_json_string):
+        try:
+            return json.loads(cesar_json_string, object_hook=Cesar.from_json)
+
+        except TypeError as e:
+            print(e)
+
+    """method to convert python object to json file"""
+
+    def to_json_file(self):
+        with open('cesar_file.json', "w") as write_file:
+            return json.dump(self, write_file, default=Cesar.to_json)
+
+    def from_json_file(cls):
+        with open('cesar_file.json', "r") as file:
```

Comment:
`file` - зарезервоване слово в пайтоні, краще назви його інакше
```suggestion
        with open('cesar_file.json', "r") as file:
```
```suggestion
        with open('cesar_file.json', "r") as f:
```